primitives/api/src/lib.rs

```diff
@@ -734,5 +734,13 @@ decl_runtime_apis! {
 	pub trait Metadata {
 		/// Returns the metadata of a runtime.
 		fn metadata() -> OpaqueMetadata;
+
+		/// Returns the metadata at a given version.
+		fn metadata_at_version(version: u32) -> Option<OpaqueMetadata>;
```

Comment:
```suggestion
		/// Returns the metadata at a given version.
		///
		/// If the given `version` isn't supported, this will return `None`.
		/// Use [`Self::metadata_versions`] to find out about supported metadata version of the runtime.
		fn metadata_at_version(version: u32) -> Option<OpaqueMetadata>;
```